Donation-platform/src/main/java/com/google/DbCalls.java

```diff
@@ -110,17 +97,144 @@ public void addUser(String userName, String email,
     userEntity.setProperty("charitiesDonatedTo", charitiesDonatedTo);
     datastore.put(userEntity);
   }
-  // Function updates trending score of a charity in the database.
-  public void updateCharityTrendingScore(Key id, Double trendingScore) throws Exception{
-    Entity charityEntity = datastore.get(id);
-    charityEntity.setProperty("trendingScore", trendingScore);
-    datastore.put(charityEntity);
+  // Function returns charity object from ID.
+  public Charity getCharityById(Key charityId) throws Exception{
+    Entity charityEntity = datastore.get(charityId);
+    Charity charityClass = setCharityClass(charityEntity);
+    return charityClass;
+  }
+  // Function gets one charity by name. Returns error if multipe charities
+  // have the same name.
+  public Charity getCharityByName(String name) throws Exception {
+    Query query =
+    new Query("Charity")
+        .setFilter(new FilterPredicate("name", FilterOperator.EQUAL, name));
+    PreparedQuery pq = datastore.prepare(query);
+    Entity result = pq.asSingleEntity();
+    Charity charityResult = setCharityClass(result);
+    return charityResult;
+  }
+  // Function returns all charities under a category tag.
+  public Collection<Charity> getCharitiesByTag(String name) throws Exception {
+    ArrayList< Charity > charityDataStore = new ArrayList < Charity > ();
+    Query query = new Query("Charity");
+    PreparedQuery results = datastore.prepare(query);
+    for (Entity entity : results.asIterable())
+        {
+            Charity charityData = setCharityClass(entity);
+            // Read the categories assigned to the charity
+            Collection<Key> categoryKeys = charityData.getCategories();
+            if (categoryKeys != null){
+              for (Key key : categoryKeys ){
+                Entity tagEntity = datastore.get(key);
+                Tag tagClass = setTagClass(tagEntity);
+                String tagName = tagClass.getName();
+                // Check to see if category contains tag searched for.
+                if (tagName.equals(name)){
+                  charityDataStore.add(charityData);
+                }
+              } 
```

Comment:
Added comment below. Avoid un-necessary declaration of variables. https://techbeacon.com/app-dev-testing/why-unnecessary-variables-are-bad-your-code

Reply:
noted.